### ISSUE TITLE
space-infix-ops: Doesnt report all the errors

### ISSUE THREAD

**Issue by gyandeeps** [eslint/eslint#3761]
**Eslint version :** 1.4.1
**Parser :** Default

**Code :**
```js
/* eslint space-infix-ops: 2 */
var foo = bar? 1: 2;
```

**Current output :**
```sh
2:14 - Infix operators must be spaced.
```

**Expected output :**
```sh
2:14 - Infix operators must be spaced.
2:17 - Infix operators must be spaced.
```

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by lo1tuma** [comment]
I think this is the expected behavior.
As far as I know, the colon in your example isn’t a separate operator (see: http://www.ecma-international.org/ecma-262/6.0/index.html#sec-conditional-operator).

**Comment by gyandeeps** [comment]
in this example   if you fix the first error (2:14) then it would start showing the second error(2:17).

**Comment by lo1tuma** [comment]
Same logic as for

```js
a+b
```

if you fix it to be `a +b` you would still have one error left.

**Comment by gyandeeps** [comment]
in your case you are talking about the same operator, in my case they are 2 different operators.

**Comment by lo1tuma** [comment]
As I understand it `? :` is only one operator (a ternary infix operator).

**Comment by nzakas** [comment]
We are also deprecating `space-infix-ops`, so differences should be discussed. https://github.com/eslint/eslint/issues/3587#issuecomment-137581672

**Comment by gyandeeps** [comment]
closing for #3587
